massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -267,6 +268,110 @@ mod tests {
+        storage.store_operations(vec![operation.clone()]);
+        let block = create_block(KeyPair::generate(), vec![operation], Slot::new(1, 0)).unwrap();
+        // store the block in storage
+        storage.store_block(block.clone());
+
+        // set our block as a final block so the message is sent
+        let mut finalized_blocks: HashMap<Slot, BlockId> = Default::default();
+        finalized_blocks.insert(block.content.header.content.slot, block.id);
+        let mut block_storage: PreHashMap<BlockId, Storage> = Default::default();
+        block_storage.insert(block.id, storage.clone());
+        controller.update_blockclique_status(
+            finalized_blocks.clone(),
+            Default::default(),
+            block_storage.clone(),
+        );
+
+        std::thread::sleep(Duration::from_millis(100));
+
+        // assert_eq!(balance, balance_expected);
+        // retrieve events emitted by smart contracts
+        let events = controller.get_filtered_sc_output_event(EventFilter {
+            start: Some(Slot::new(0, 1)),
+            end: Some(Slot::new(20, 1)),
+            ..Default::default()
+        });
+        // match the events
+        assert!(!events.is_empty(), "One event was expected");
+        let address = events[0].clone().data;
+        // Call the function test of the smart contract
+        let operation = create_call_sc_operation(
+            &keypair,
+            10000000,
+            Amount::from_str("0").unwrap(),
+            Amount::from_str("10").unwrap(),
+            Address::from_str(&address).unwrap(),
+            String::from("test"),
+            address.as_bytes().to_vec(),
+        )
+        .unwrap();
+        // Init new storage for this block
+        let mut storage = Storage::create_root();
+        storage.store_operations(vec![operation.clone()]);
+        let block = create_block(KeyPair::generate(), vec![operation], Slot::new(2, 0)).unwrap();
+        // store the block in storage
+        storage.store_block(block.clone());
+        // set our block as a final block so the message is sent
+        let mut finalized_blocks: HashMap<Slot, BlockId> = Default::default();
+        finalized_blocks.insert(block.content.header.content.slot, block.id);
+        let mut block_storage: PreHashMap<BlockId, Storage> = Default::default();
+        block_storage.insert(block.id, storage.clone());
+        controller.update_blockclique_status(
+            finalized_blocks,
+            Default::default(),
+            block_storage.clone(),
+        );
+        std::thread::sleep(Duration::from_millis(100));
+        // Get the events that give us the gas usage (refer to source in ts) without fetching the first slot because it emit a event with an address.
+        let events = controller.get_filtered_sc_output_event(EventFilter {
+            start: Some(Slot::new(2, 0)),
+            ..Default::default()
+        });
+        println!("events: {:#?}", events);
```

Comment:
Should test the message content (and check for the coin amount)?

Reply:
Done